### ISSUE TITLE
TonApiClient 2.0 POST method not working

### ISSUE THREAD

**Issue by allen-hsu** [tonkeeper/tonapi-js#206]
as title, forgot to pass the method to req, always use 'GET', so when call 'POST' function, will get 405 issue
![截圖 2024-11-23 上午10 27 27](https://github.com/user-attachments/assets/dfdfc506-e031-4568-8ded-b49538e5ff32)

**Comment by allen-hsu** [comment]
and your prepareResponse will convert the correct error information into the wrong format because of

```
async function prepareResponse(promise, orSchema) {
    return await promise
        .then(obj => prepareResponseData(obj, orSchema))
        .catch(async (error) => {
        let errorMessage;
        if (error &&
            typeof error === 'object' &&
            'json' in error &&
            typeof error.json === 'function') {
            try {
                const errorJson = await error.json();
                errorMessage =
                    typeof errorJson === 'string'
                        ? errorJson
                        : errorJson?.error || 'Unknown error';
            }
            catch (jsonParseError) {
                if (jsonParseError instanceof Error) {
                    errorMessage = `Failed to parse error response: ${jsonParseError.message}`;
                }
                else {
                    errorMessage = 'Failed to parse error response: Unknown parsing error';
                }
            }
        }
        else if (error instanceof Error) {
            errorMessage = error.message || 'Unknown error without JSON';
        }
        else {
            errorMessage = 'Unknown error occurred';
        }
        throw new Error(errorMessage, { cause: error });
    });
    ```

**Comment by mois-ilya** [comment]
@allen-hsu  Hi! The method is passed in requestParams and handled correctly in the current implementation. I wrote a test for this case and couldn’t reproduce the issue you described.

Could you provide steps to reproduce the issue? It might be related to specific conditions in your environment or configuration.

---

error information information fixed in [0.3.0-beta.0](https://www.npmjs.com/package/@ton-api/client/v/0.3.0-beta.0) now

**Comment by mois-ilya** [comment]
@allen-hsu  Hi! The method is passed in requestParams and handled correctly in the current implementation. I [wrote a test](https://github.com/tonkeeper/tonapi-js/commit/5829fc5652d7cb89b08117fdf06cf7508bf9d150) for this case and couldn’t reproduce the issue you described.

Could you provide steps to reproduce the issue? It might be related to specific conditions in your environment or configuration.

---

error information information fixed in [0.3.0-beta.0](https://www.npmjs.com/package/@ton-api/client/v/0.3.0-beta.0) now

**Comment by mois-ilya** [comment]
Closed the issue as there is no explanation of the problem
